cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -98,55 +107,31 @@ def test_webmks_vm_console(request, appliance, provider, vm_obj,
         # Enter Password:
         vm_console.send_keys("{}\n".format(console_vm_password))
 
-        result_after_login = vm_ssh_client.run_command("who --count", ensure_user=True)
+        result_after_login = ssh_client.run_command("who --count", ensure_user=True)
         # Number of users before login would be 0 and after login would be 180
         # If below assertion would fail result_after_login is also 0, denoting login failed
         assert (result_before_login.output.split('=')[-1].strip() <
              result_after_login.output.split('=')[-1].strip()), "Login Failed"
 
-    # This regex can find if there is a word 'login','password','incorrect' present in
-    # text, irrespective of its case
-    regex_for_login_password = re.compile(r'\blogin\b | \bpassword\b| \bincorrect\b',
-     flags=re.I | re.X)
-
-    def _validate_login():
-        """
-        Try to read what is on present on the last line in console.
-
-        If it is word 'login', enter username, if 'password' enter password, in order
-        to make the login successful
-        """
-        if vm_console.find_text_on_screen(text_to_find='login', current_line=True):
-            vm_console.send_keys(console_vm_username)
-
-        if vm_console.find_text_on_screen(text_to_find='Password', current_line=True):
-            vm_console.send_keys("{}\n".format(console_vm_password))
-        # if the login attempt failed for some reason (happens with RHOS-cirros),
-        # last line of the console will contain one of the following words:
-        # [login, password, incorrect]
-        # if so, regex_for_login_password will find it and result will not be []
-        # .split('\n')[-1] splits the console text on '\n' & picks last item of resulting list
-        result = regex_for_login_password.findall(vm_console.get_screen_text().split('\n')[-1])
-        return result == []
+        logger.info("Wait to get the '$' prompt")
 
-    # if _validate_login() returns True, it means we did not find any of words
-    # [login, password, incorrect] on last line of console text, which implies login success
-    wait_for(func=_validate_login, timeout=300, delay=5)
+        vm_console.wait_for_text(text_to_find=provider.data.templates.get('console_template')
+            ['prompt_text'], timeout=200)
 
-    logger.info("Wait to get the '$' prompt")
-
-    vm_console.wait_for_text(text_to_find=provider.data.templates.get('console_template')
-        ['prompt_text'], timeout=200)
-
-    with ssh.SSHClient(hostname=vm_obj.ip_address, username=console_vm_username,
-            password=console_vm_password) as vm_ssh_client:
         # create file on system
         vm_console.send_keys("touch blather\n")
-        wait_for(func=lambda: vm_ssh_client.run_command("ls blather", ensure_user=True) == 0,
+        wait_for(func=lambda: ssh_client.run_command("ls blather", ensure_user=True) == 0,
```

Comment:
**Optional, Recommended**

Use `func_args`, `func_kwargs`, and `fail_condition` arguments to wait_for instead of using lambda and trying to create a false in `func`.